File: javaPOO/POOAula07/src/aula07poo/Lutador.java
package aula07poo;
public class Lutador {
    private String nome,categoria,nacionalidade;
    private int idade;
    private float peso, altura;
    private int vitoria,derrota,empate;
    
    
    public void apresentar(){
        System.out.println("--------------------------------");
        System.out.println("Apresentando o Lutador "+this.getNome());
        System.out.println("Nascionalidade: "+this.getNacionalidade());
        System.out.println("Idade: "+ this.getIdade()+" anos");
        System.out.println("Pesando: "+this.getPeso()+"Kg");
        System.out.println(this.getVitoria()+" Vitórias");
        System.out.println(this.getDerrota()+" Derrotas");
        System.out.println(this.getEmpate()+" Empates");
    }
    public void status(){
        System.out.println(this.getNome()+" é um peso "+this.getCategoria());
        System.out.println("Ganhou "+this.getVitoria()+" vezes");
        System.out.println("Perdeu "+this.getDerrota()+" vezes");
        System.out.println("Empatou "+this.getEmpate()+" vezes");
    }
    public void ganharLuta(){
        this.setVitoria(this.getVitoria()+1);
    }
    public void perderLuta(){
        this.setDerrota(this.getDerrota()+1);
    }
    public void empatarLuta(){
        this.setEmpate(this.getEmpate()+1);
    }

    public Lutador(String nome, String nacionalidade, int idade,float altura, float peso, int vitoria, int derrota, int empate) {
        this.nome = nome;
        this.nacionalidade = nacionalidade;
        this.idade = idade;
        this.altura = altura;
        this.setPeso(peso);
        this.vitoria = vitoria;
        this.derrota = derrota;
        this.empate = empate;
    }

    public String getNome() {
        return nome;
    }

    public void setNome(String nome) {
        this.nome = nome;
    }

    public String getCategoria() {
        return categoria;
    }

    private void setCategoria() {
        if(this.getPeso()< 52.2){
            this.categoria = "Invalido";
        }else if(this.peso<= 70.3){
            this.categoria = "Leve";
        }else if(this.peso<= 83.9){
            this.categoria = "Médio";
        }else if(this.peso<= 120.2){
            this.categoria = "Pesado";
        }else{
            this.categoria = "Invalido";
        }
    }

    public String getNacionalidade() {
        return nacionalidade;
    }

    public void setNacionalidade(String nacionalidade) {
        this.nacionalidade = nacionalidade;
    }

    public int getIdade() {
        return idade;
    }

    public void setIdade(int idade) {
        this.idade = idade;
    }

    public float getPeso() {
        return peso;
    }

    public void setPeso(float peso) {
        this.peso = peso;
        this.setCategoria();
    }

    public int getVitoria() {
        return vitoria;
    }

    public void setVitoria(int vitoria) {
        this.vitoria = vitoria;
    }

    public int getDerrota() {
        return derrota;
    }

    public void setDerrota(int derrota) {
        this.derrota = derrota;
    }

    public int getEmpate() {
        return empate;
    }

    public void setEmpate(int empate) {
        this.empate = empate;
    }
    
}

File: phpPOO/POOAula07/Lutador.php
<?php

class Lutador {
    private $nome;
    private $nacionalidade;
    private $idade,$altura,$peso;
    private $categoria, $vitora, $derrota, $empate;
    
    function apresentar(){
        print "<p>------------------------</p>";
        print "<p>O lutador: ".$this->getNome();
        print " veio de ".$this->getNacionalidade();
        print " tem ".$this->getIdade()." anos e pesa ". $this->getPeso()."Kg";
        print "<br>Ele tem " .$this->getVitora()." vitórias, ";
        print $this->getDerrota()." derrotas e ".$this->getEmpate()." empates.</p>";
        
    }
    function status(){
        print "<p>----------------------</p>";
        print "<p>" . $this->getNome() . " é um peso " . $this->getCategoria();
        print " e ja ganhou ". $this->getVitora()." vezes,";
        print " perdeu ". $this->getDerrota()." vezes e ";
        print " empatou ".$this->getEmpate()." vezes";
    }
    function ganharLuta(){
        $this->setVitora($this->getVitora()+1);
    }
    function perderLuta(){
        $this->setDerrota($this->getDerrota()+1);
    }
    function empatarLuta(){
        $this->setEmpate($this->getEmpate()+1);
    }
    public function __construct($nome, $nacionalidade, $idade, $altura, $peso, $vitora, $derrota, $empate) {
        $this->nome = $nome;
        $this->nacionalidade = $nacionalidade;
        $this->idade = $idade;
        $this->altura = $altura;
        $this->setPeso($peso); 
        $this->vitora = $vitora;
        $this->derrota = $derrota;
        $this->empate = $empate;
    }
    public function getNome() {
        return $this->nome;
    }

    public function getNacionalidade() {
        return $this->nacionalidade;
    }

    public function getIdade() {
        return $this->idade;
    }

    public function getAltura() {
        return $this->altura;
    }

    public function getPeso() {
        return $this->peso;
    }

    public function getCategoria() {
        return $this->categoria;
    }

    public function getVitora() {
        return $this->vitora;
    }

    public function getDerrota() {
        return $this->derrota;
    }

    public function getEmpate() {
        return $this->empate;
    }

    public function setNome($nome): void {
        $this->nome = $nome;
    }

    public function setNacionalidade($nacionalidade): void {
        $this->nacionalidade = $nacionalidade;
    }

    public function setIdade($idade): void {
        $this->idade = $idade;
    }

    public function setAltura($altura): void {
        $this->altura = $altura;
    }

    public function setPeso($peso): void {
        $this->peso = $peso;
        $this->setCategoria();
    }

    public function setCategoria(): void {
        if($this->getPeso()<52.2){
            $this->categoria = "Invalido";
        }elseif($this->getPeso()<=70.3){
            $this->categoria = "Leve";
        }elseif($this->getPeso()<=83.9){
            $this->categoria = "Médio";
        }elseif($this->getPeso()<= 120.2){
            $this->categoria = "Pesado";
        }else{
            $this->categoria = "Invalido";
        }
    }

    public function setVitora($vitora): void {
        $this->vitora = $vitora;
    }

    public function setDerrota($derrota): void {
        $this->derrota = $derrota;
    }

    public function setEmpate($empate): void {
        $this->empate = $empate;
    }



}
